tnoodle-ui/src/main/api/tnoodle.socket.js

```diff
@@ -0,0 +1,82 @@
+import { tNoodleBackend } from "./tnoodle.api";
+
+export class ScrambleClient {
+    constructor(onHandshake, onProgress) {
+        this.onHandshake = onHandshake;
+        this.onProgress = onProgress;
+
+        this.state = SCRAMBLING_STATES.IDLE;
+
+        this.contentType = null;
+        this.resultPayload = null;
+    }
+
+    loadScrambles(endpoint, payload, targetMarker) {
+        let that = this;
+
+        return new Promise(function (resolve, reject) {
```

Comment:
I think that if you use
```
(resolve, reject) => {
...
}

```
you'll be able to dismiss the use of `that` and keep using `this`. Arrow functions allow this.

Reply:
And if you're going to touch it, maybe you could also take care of this other one

https://github.com/thewca/tnoodle/blame/114715cd0410b83b19e92464ff6dc52e2abde8d5/tnoodle-ui/src/main/components/Interceptor.jsx#L15